src/java/org/apache/cassandra/index/sai/utils/IndexFileUtils.java

```diff
@@ -55,13 +55,29 @@ public class IndexFileUtils
                                                                                               .finishOnClose(true)
                                                                                               .build();
 
-    public static final IndexFileUtils instance = new IndexFileUtils();
+    private static final IndexFileUtils instance = new IndexFileUtils(defaultWriterOption);
+    private static IndexFileUtils overrideInstance = null;
```

Comment:
It's a bit nitpicking, but is there any particular reason we use a combination of two variables instead of just marking the first one non-final and updating it in place? Then we wouldn't need to change all the stuff from `.instance` to `.instance()` (which is fine, but I'm worried it increases the conflict area in any merges we do e.g. wit apache). Is there any downside of using a single field?

Reply:
Not sure why it was done this way, this comes from [a previous set of changes in `main-5.0`](https://github.com/datastax/cassandra/commit/047d33c904e6082708000698b4342d236b427f71#diff-6a43d8be03e631756ea0fd2a6fd92b6dd106ed1450ab6cabf40dd86d551bb47d). Agree on the single variable. Regarding having getters and setters, the only reason that occurs to me is perhaps having less visibility in the setter? In any case, I have changed it to the simplest approach.

By the way, regarding Apache we have CASSANDRA-19961 to update this there too, albeit in that case the problem is that some inputs are not being closed.